src/action/payment.js

```diff
@@ -68,8 +68,13 @@ class PaymentAction {
       const request = await this._grpc.sendCommand('decodePayReq', {
         pay_req: invoice.replace(PREFIX_URI, ''),
       });
+      const fee = await this.estimateLightningFee({
+        destination: request.destination,
+        satAmt: request.num_satoshis,
+      });
       payment.amount = toAmount(parseSat(request.num_satoshis), settings.unit);
       payment.note = request.description;
+      payment.fee = toAmount(parseSat(fee), settings.unit);
       return true;
```

Comment:
This would freeze the UI if `queryRoutes` takes a few seconds. Have you tested this with Yalls and Starblocks?

Reply:
Ya that's a good point, I did test with Yalls and Starblocks and it was fine but I'm adding a wait screen if `queryRoutes` takes more than 500ms.